src/overcooked_ai_py/mdp/overcooked_env.py

```diff
@@ -664,6 +665,136 @@ def proportion_stuck_time(trajectories, agent_idx, stuck_time=3):
         return stuck_matrix
 
 
+class OvercookedEnv_PettingZoo(gymnasium.Wrapper):
+    """
+    Create environment for gymnasium for pettingZoo
+    """
+
+    def step(self, joint_action):
+        obs, reward, done, info = super().step(joint_action)
+        # from the doc, this is true only if a pre set goal is achieved, which we don't have in this environment
+        # the only termination condition is the preset max steps in the environment
+        terminated = False
+        truncated = done
+        return obs, reward, terminated, truncated, info
+
+
+from pettingzoo.utils.env import ParallelEnv
+
+from overcooked_ai_py.agents.agent import AgentPair
+
+
+class OvercookedEnv_Pet(ParallelEnv):
+    def __init__(self, base_env, agents):
+        """
+        base_env: OvercookedEnv
+        featurize_fn(mdp, state): fn used to featurize states returned in the 'both_agent_obs' field
+
+        Example creating a gym env:
+
+        mdp = OvercookedGridworld.from_layout_name("asymmetric_advantages")
+        base_env = OvercookedEnv.from_mdp(mdp, horizon=500)
```

Comment:
Ah I see there’s a horizon argument so maybe that’s the done signal you used earlier for truncation

Reply:
Yeah that's correct